Client/Frontend/Browser/TabManager.swift

```diff
@@ -379,9 +381,41 @@ extension TabManager {
 }
 
 extension TabManager : WKNavigationDelegate {
+    func webView(webView: WKWebView, didStartProvisionalNavigation navigation: WKNavigation!) {
+        if tabsNetworkActivity[webView] == nil {
+            tabsNetworkActivity[webView] = false
+        }
+        if tabsNetworkActivity[webView] == false {
+            tabsNetworkActivity[webView] = true
+            incrementNetworkActivityCount()
+        }
+    }
+
     func webView(webView: WKWebView, didFinishNavigation navigation: WKNavigation!) {
+        tabsNetworkActivity[webView] = false
+        decrementNetworkActivityCount()
+
         storeChanges()
     }
+
+    func webView(webView: WKWebView, didFailNavigation navigation: WKNavigation!, withError error: NSError) {
+        tabsNetworkActivity[webView] = false
+        decrementNetworkActivityCount()
+    }
+
+    func incrementNetworkActivityCount() {
```

Comment:
Yes I need to take that into account, the delegate methods are not called after closing the tab. But for this I need to keep the network state of each tab (active/inactive) in memory so I can know if I need to decrement the counter after closing a tab.